config/src/lib.rs

```diff
@@ -1187,39 +1187,205 @@ impl Config {
 
         dir_size_recursive(fs::read_dir(chain_path)?)
     }
+
+    fn fix_toml_for_profiles(toml_path: impl AsRef<Path>, profiles: &[Profile]) {
+        // parse toml
+        let toml_path = toml_path.as_ref();
+        let mut toml_doc: toml_edit::Document = match std::fs::read_to_string(toml_path) {
+            Ok(toml_str) => match toml_str.parse() {
+                Ok(value) => value,
+                Err(err) => {
+                    config_warn!("Invalid TOML at {}: {}", toml_path.display(), err);
+                    return
+                }
+            },
+            Err(err) => {
+                config_warn!("Could not read TOML at {}: {}", toml_path.display(), err);
+                return
+            }
+        };
```

Comment:
this could potentially be reused down the line, so probably worth it to make this a stand-alone helper function